packages/solid/src/reactive/observable.ts

```diff
@@ -29,16 +29,29 @@ export function observable<T>(input: Accessor<T>) {
       if (!(observer instanceof Object) || observer == null) {
         throw new TypeError("Expected the observer to be an object.");
       }
-      const handler = "next" in observer ? observer.next.bind(observer) : observer;
-      let complete = false;
-      createComputed(() => {
-        if (complete) return;
-        const v = input();
-        untrack(() => handler(v));
-      });
+
+      const handler = typeof observer === 'function'
+        ? observer
+        : observer.next?.bind(observer);
```

Comment:
Small nit. We don't use optional chaining in Solid core as it breaks some of the platforms we support. And we won't transpile the source code. Don't always trust it to do the most performant thing.

Reply:
Fixed 👍. Was able to make all the changes inside a [Codespace](https://github.com/features/codespaces) which is kinda cool.